File: src/main/java/org/flossware/civilization/model/TradeRoute.java
package org.flossware.civilization.model;

import java.util.List;

/**
 * Represents a trade connection between civilizations.
 */
public record TradeRoute(
    String from,
    String to,
    List<String> goods,
    double volume,
    double tariff
) {
    public TradeRoute {
        if (volume < 0) {
            throw new IllegalArgumentException("Trade volume cannot be negative");
        }
        if (tariff < 0 || tariff > 1) {
            throw new IllegalArgumentException("Tariff must be in [0, 1]");
        }
        goods = List.copyOf(goods);
    }
}

File: mobile/shared/src/commonMain/kotlin/org/flossware/civilization/model/TradeRoute.kt
package org.flossware.civilization.model

import kotlinx.serialization.Serializable

@Serializable
data class TradeRoute(
    val from: String,
    val to: String,
    val goods: List<String>,
    val volume: Double,
    val tariff: Double
) {
    init {
        require(volume >= 0) { "Trade volume cannot be negative" }
        require(tariff in 0.0..1.0) { "Tariff must be in [0, 1]" }
    }
}
